File: C++/Easy/Accepted/check-if-all-characters-have-equal-number-of-occurrences_9.cpp
class Solution {
public:
    bool areOccurrencesEqual(string s) {
        
        unordered_map<char, int> m;
        for (auto c : s) {
            m[c]++;
        }
        int n = m.size();
        int count = m[s[0]];
        for (auto [c, i] : m) {
            if (i != count) {
                return false;
            }
        }
        return true;
    }
};

File: Java/Easy/Accepted/check-if-all-characters-have-equal-number-of-occurrences_9.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        
        int[] count = new int[26];
        for (char c : s.toCharArray()) {
            count[c - 'a']++;
        }
        int prev = 0;
        for (int n : count) {
            if (n != 0) {
                if (prev == 0) {
                    prev = n;
                } else if (prev != n) {
                    return false;
                }
            }
        }
        return true;
    }
}